src/frontend/context/MottakerTypeContext.tsx

```diff
@@ -0,0 +1,211 @@
+import { feil, ok, useFelt, useSkjema, Valideringsstatus } from '@navikt/familie-skjema';
+import type { Ressurs } from '@navikt/familie-typer';
+import { RessursStatus } from '@navikt/familie-typer';
+
+import useSakOgBehandlingParams from '../hooks/useSakOgBehandlingParams';
+import { BehandlingSteg } from '../typer/behandling';
+import type { IBehandling } from '../typer/behandling';
+import { FagsakType } from '../typer/fagsak';
+import type { IRegistrerInstitusjonOgVerge } from '../typer/institusjon-og-verge';
+import type { IPersonInfo } from '../typer/person';
+import { hentAlder, kunSiffer } from '../utils/formatter';
+import { hentFrontendFeilmelding } from '../utils/ressursUtils';
+import { identValidator } from '../utils/validators';
+import { useBehandling } from './behandlingContext/BehandlingContext';
+import { useFagsakRessurser } from './FagsakContext';
+
+const [MottakerTypeProvider, useMottakerType] = createUseContext(
+    ({ åpenBehandling }: { åpenBehandling: IBehandling }) => {
+        const { erLesevisning, settÅpenBehandling } = useBehandling();
+        const { minimalFagsak } = useFagsakRessurser();
+        const { fagsakId } = useSakOgBehandlingParams();
+        const navigate = useNavigate();
+        const { request } = useHttp();
+        const [feilMelding, settFeilMelding] = useState<string | undefined>('');
+        const lesevisning = () =>
+            erLesevisning() ||
+            åpenBehandling?.steg !== BehandlingSteg.REGISTRERE_INSTITUSJON_OG_VERGE;
+
+        const fagsakType: { [key: string]: string } = {
+            data:
+                minimalFagsak.status === RessursStatus.SUKSESS ? minimalFagsak.data.fagsakType : '',
+            feilmelding:
+                minimalFagsak.status !== RessursStatus.SUKSESS
+                    ? hentFrontendFeilmelding(minimalFagsak) || 'Ukjent feil ved henting av fagsak'
+                    : '',
+        };
+        const institusjon = useFelt<string | undefined>({
+            verdi: undefined,
+        });
+        const { skjema, onSubmit } = useSkjema<
+            {
+                fødselsnummer: string;
+                institusjon: string | undefined;
+                navn: string | undefined;
+                adresse: string;
+                postnummer: string;
+                sted: string;
+            },
+            IBehandling
+        >({
+            felter: {
+                fødselsnummer: useFelt<string>({
+                    verdi: '',
+                    avhengigheter: { feilMelding },
+                    valideringsfunksjon: (felt, avhengigheter) => {
+                        if (avhengigheter?.feilMelding) {
+                            return feil(felt, avhengigheter?.feilMelding);
+                        } else {
+                            return felt.verdi === '' ? ok(felt) : identValidator(felt);
+                        }
+                    },
+                }),
+                institusjon: institusjon,
```

Comment:
I do not quite remember the skjema stuff, but other felter are defined as useFelt<>, why here it is the plain variable institusjon?

Reply:
institusjon is defined as useFelt<> as well, just not inline like the others since it also is referenced as a dependency (avhengighet) in the field below to decide wether it should be optional or not.